java/core/src/main/java/com/microsoft/bond/Blob.java

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.bond;
+
+import java.util.Arrays;
+
+/**
+ * Provides implementation for the Bond blob type as a simple wrapper around {@code byte []}.
+ */
+public final class Blob {
+
+    /**
+     * Creates a new Blob instance
+     * @return new Blob instance with an empty {@code byte []}
+     */
+    public Blob() {
+        this.data = new byte[0];
+    }
+
+    /**
+     * Creates a new Blob instance wrapping an existing {@code byte []}
+     * @param data
+     * @return new Blob instance wrapping the data array
+     * @throws IllegalArgumentException if the argument is null
+     */
+    public Blob(byte[] data) {
+        if (data == null) {
+            throw new IllegalArgumentException("Argument 'data' must not be null");
+        }
+        this.data = data;
+    }
+
+    /**
+     * Compare for equality by contents of underlying {@code byte []}
+     * @param other object to compare against
+     * @return {@code true} if object is also a Blob and the underlying arrays have equivalent contents
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true;
+        }
```

Comment:
Not exactly. The parameters to Array.equals() are the arrays -- this is the optimization at the level of Object.